Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -5,24 +5,214 @@
 import Foundation
 import UIKit
 
-private let StatusBarHeight = 20
+private let StatusBarHeight = CGFloat(20)
+private let TextBoxHeight = CGFloat(25.0)
+private let Margin = CGFloat(10)
+private let ToolbarHeight: CGFloat = 44
+private let CellHeight = TextBoxHeight * 4
+
+@objc
+class TransitionOptions {
+    var data = [String: AnyObject]()
+
+    subscript(key: String) -> AnyObject? {
+        get {
+            return data[key]
+        }
+        set(newValue) {
+            data[key] = newValue
+        }
+    }
+}
+
+@objc
+protocol Transitionable : class {
+    func hideTransition(opts: TransitionOptions, vc: UIViewController)
```

Comment:
opts -> options, vc -> viewController. iOS convention is to use full names for things.